avocado/plugins/messages.py

```diff
@@ -0,0 +1,123 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2021
+# Authors: Jan Richter <jarichte@redhat.com>
+
+"""
+Message resolver for runner messages
+"""
+
+import abc
+import os
+
+
+class MessageProcessor:
+    """Entry point for handling all types of messages"""
+
+    def __init__(self):
+        self._handlers = [StartMessageHandler(),
+                          RunningMessageProcessor(),
+                          FinishMessageHandler()]
+
+    def process_message(self, message, task, job):
```

Comment:
Could you please add a docstring here?

Reply:
Yes in the final version I am planing to add docstrings.